### PR TITLE
Fix flytescheduler init check for authenticated admins

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
# TL;DR

Flytescheduler checks for the health of the admin before bringing itself up.
Earlier these checks were done by creating health client in the scheduler and passing some set of grpc dial options.
These grpc dial options were insufficient in cases of authenticated admins.

This change uses the HealthClient as part of the clientset which underneath takes care of setting the right dial options based on the admin configuration.

* Also fixes the permission issues with the secrets folder creation due to introduction of new flyte user and flyte group

Testing done : 
Using the same in flytescheduler-check 

```
flytescheduler precheck --config admin.yaml 
Using config file:  [admin.yaml]
{"json":{},"level":"error","msg":"failed to initialize token source provider. Err: failed to fetch auth metadata. Error: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:8089: connect: connection refused\"","ts":"2021-11-05T15:41:43+05:30"}
{"json":{},"level":"warning","msg":"Starting an unauthenticated client because: can't create authenticated channel without a TokenSourceProvider","ts":"2021-11-05T15:41:43+05:30"}
{"json":{},"level":"info","msg":"Initialized Admin client","ts":"2021-11-05T15:41:43+05:30"}
{"json":{},"level":"error","msg":"Attempt failed due to rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:8089: connect: connection refused\"","ts":"2021-11-05T15:41:43+05:30"}
{"json":{},"level":"info","msg":"Initialized Admin client","ts":"2021-11-05T15:41:44+05:30"}
{"json":{},"level":"error","msg":"Attempt failed due to rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:8089: connect: connection refused\"","ts":"2021-11-05T15:41:45+05:30"}
{"json":{},"level":"info","msg":"Initialized Admin client","ts":"2021-11-05T15:41:47+05:30"}
{"json":{},"level":"error","msg":"Attempt failed due to rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:8089: connect: connection refused\"","ts":"2021-11-05T15:41:47+05:30"}
{"json":{},"level":"info","msg":"Initialized Admin client","ts":"2021-11-05T15:41:51+05:30"}
{"json":{},"level":"error","msg":"Attempt failed due to rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp [::1]:8089: connect: connection refused\"","ts":"2021-11-05T15:41:51+05:30"}
{"json":{},"level":"info","msg":"Initialized Admin client","ts":"2021-11-05T15:42:00+05:30"}
{"json":{},"level":"info","msg":"Health check response is status:SERVING","ts":"2021-11-05T15:42:00+05:30"}
{"json":{},"level":"info","msg":"Health check passed, Flyteadmin is up and running","ts":"2021-11-05T15:42:00+05:30"}
```

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [X] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1753

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -31,8 +31,10 @@ COPY --from=builder /artifacts /bin
 
 # Ensure the latest CA certs are present to authenticate SSL connections.
 RUN apk --update add ca-certificates
+RUN mkdir /etc/secrets
 
 RUN addgroup -S flyte && adduser -S flyte -G flyte
+RUN chown -R flyte:flyte /etc/secrets
```
[the remark this answers is not in the document]
Comment (pmahindrakar-oss) [reply Dockerfile]: Have removed the docker file changes from this PR to unblock the current one.

File: cmd/scheduler/entrypoints/precheck.go [hunk 2]
```diff
@@ -3,83 +3,61 @@ package entrypoints
...
-			grpc.WithUserAgent("grpc_health_probe"),
-			grpc.WithBlock(),
-			grpc.WithInsecure(),
-		}
 		ctx := context.Background()
-		config := adminClient.GetConfig(ctx)
 
-		err := retry.Do(
+		// Do maximum of 30 retries on failures with constant backoff factor
+		opts := wait.Backoff{Duration: 3000, Factor: 2.0, Steps: 30, Jitter: 0.1}
```
Comment (EngHabu) [inline cmd/scheduler/entrypoints/precheck.go:28]: Can you move these to a config with these numbers as the default?

Comment (pmahindrakar-oss) [reply cmd/scheduler/entrypoints/precheck.go]: Done.

**Review by EngHabu — APPROVED** [review]
